src/inc/til/rectangle.h

```diff
@@ -13,9 +13,84 @@ class RectangleTests;
 
 namespace til // Terminal Implementation Library. Also: "Today I Learned"
 {
+    class _rectangle_const_iterator
+    {
+    public:
+        constexpr _rectangle_const_iterator(point topLeft, point bottomRight) :
+            _topLeft(topLeft),
+            _bottomRight(bottomRight),
+            _current(topLeft)
+        {
+        }
+
+        constexpr _rectangle_const_iterator(point topLeft, point bottomRight, point start) :
+            _topLeft(topLeft),
+            _bottomRight(bottomRight),
+            _current(start)
+        {
+        }
+
+        _rectangle_const_iterator& operator++()
+        {
+            ptrdiff_t nextX;
+            THROW_HR_IF(E_ABORT, !::base::CheckAdd(_current.x(), 1).AssignIfValid(&nextX));

```

Comment:
I don't really want to fix this. The only way I can think of this happening is if you have a 0-width rectangle all the way over at X = ptrdiff_t max.

I don't super want to add some sort of check that prevents the E_ABORT throw here because it would then waste everyone's time in the 99.9999999999999999999% of circumstances where we won't be at that position and the 100% of circumstances where we aren't supposed to be at that position.